### PR TITLE
docker/registry: Make CertsDir configurable.

### PR DESCRIPTION
**Pull request by vosst**
On linux hosts, it would be helpful if certificates for
accessing a remote registry could be configured at runtime
for specific setups. One example are snaps and third-parties
who want to work with their own registries.

### CODE REVIEW

**Comment by vosst** [comment]
I tried to track down the test failure on Windows, but failed to reproduce the Jenkins setup. Do you have instructions available for debugging test failures?

File: registry/config_unix.go [hunk 1]
```diff
@@ -3,12 +3,22 @@
 package registry
 
 import (
+	"os"
+
 	"github.com/spf13/pflag"
 )
 
+func initCertsDir() string {
+	dir := os.Getenv("DOCKER_REGISTRY_CERTS_DIR")
```
Comment (cpuguy83) [inline registry/config_unix.go]: Definitely no.
Daemon flag would be much better.

**Review by cpuguy83 — CHANGES_REQUESTED** [review]

File: registry/config_unix.go (hunk 1 above)
Comment (vosst) [inline registry/config_unix.go]: Thanks, applied your review. Let me know if the name of the flag works, or if you would prefer something less clunky.

**Comment by thaJeztah** [comment]
ping @cyli @diogomonica PTAL as well

**Comment by cpuguy83** [comment]
@vosst Thanks for the update. It looks like you have the flag but it's not wired up to anything currently.

**Comment by thaJeztah** [comment]
ping @vosst are you still working on this?

**Comment by ehazlett** [comment]
ping @vosst

**Comment by thaJeztah** [comment]
closing because there was no activity for a long time

**Pull request closed without merging**